tcms/core/ajax.py

```diff
@@ -115,23 +115,24 @@ def versions(self):
     objects = Objects(request=request, product_id=request.GET.get('product_id'))
     obj = getattr(objects, request.GET.get('info_type'), None)
 
-    if obj:
-        if request.GET.get('format') == 'ulli':
-            field = request.GET.get('field', 'name')
-            response_str = '<ul>'
-            for obj_value in obj().values(field):
-                response_str += '<li>' + obj_value.get(field, None) + '</li>'
-            response_str += '</ul>'
-            return HttpResponse(response_str)
-
-        return HttpResponse(serializers.serialize(
-            request.GET.get('format', 'json'),
-            obj(),
-            fields=('name', 'value')
-        ))
-
-    return HttpResponse('Unrecognizable infotype')
+    if not obj:
+        return HttpResponse('Unrecognizable infotype')
 
+    if request.GET.get('format') == 'ulli':
+        field = request.GET.get('field', 'name')
+
+        response_str = '<ul>'
+        for obj_value in obj().values(field):
+            response_str += '<li>' + obj_value.get(field, None) + '</li>'
+        response_str += '</ul>'
+
+        return HttpResponse(response_str)
+
+    return HttpResponse(serializers.serialize(
+        request.GET.get('format', 'json'),
```

Comment:
I think you can safely hardcode 'json' as the serialization format here. I don't think we support anything else either way so if the value is actually different than 'json' this will most likely fail.